### PR TITLE
ISPN-6726 ContinuousQueryListener.resultJoining with the old value

### PR DESCRIPTION
**Pull request by anistor**
https://issues.jboss.org/browse/ISPN-6726

### CODE REVIEW

File: client/hotrod-client/src/test/java/org/infinispan/client/hotrod/event/EmbeddedCompatContinuousQueryTest.java [hunk 1]
```diff
@@ -151,6 +157,108 @@ public void testContinuousQuery() {
...
+      remoteCache.clear();
+      user1.setAge(21);
+      user2.setAge(22);
+      remoteCache.put("expiredUser1", user1, 5, TimeUnit.MILLISECONDS);
+      remoteCache.put("expiredUser2", user2, 5, TimeUnit.MILLISECONDS);
+
+      expectElementsInQueue(joined, 2);
+      expectElementsInQueue(left, 0);
+
+      TestingUtil.sleepThread(60);
```
Comment (slaskawi) [inline client/hotrod-client/src/test/java/org/infinispan/client/hotrod/event/EmbeddedCompatContinuousQueryTest.java]: Can we remove this?

File: client/hotrod-client/src/test/java/org/infinispan/client/hotrod/event/RemoteContinuousQueryTest.java [hunk 2]
```diff
@@ -136,6 +143,108 @@ public void testContinuousQuery() {
...
+      remoteCache.clear();
+      user1.setAge(21);
+      user2.setAge(22);
+      remoteCache.put("expiredUser1", user1, 5, TimeUnit.MILLISECONDS);
+      remoteCache.put("expiredUser2", user2, 5, TimeUnit.MILLISECONDS);
+
+      expectElementsInQueue(joined, 2);
+      expectElementsInQueue(left, 0);
+
+      TestingUtil.sleepThread(60);
```
Comment (slaskawi) [inline client/hotrod-client/src/test/java/org/infinispan/client/hotrod/event/RemoteContinuousQueryTest.java]: Can we remove it?

File: core/src/main/java/org/infinispan/marshall/core/ExternalizerTable.java [hunk 3]
```diff
@@ -67,7 +67,6 @@
 import org.infinispan.filter.NullValueConverter;
 import org.infinispan.functional.impl.EntryViews;
 import org.infinispan.functional.impl.MetaParams;
-import org.infinispan.functional.impl.Params;
```
Comment (slaskawi) [inline core/src/main/java/org/infinispan/marshall/core/ExternalizerTable.java]: Probably unnecessary change

File: core/src/test/java/org/infinispan/marshall/core/JBossMarshallerTest.java [hunk 4]
```diff
@@ -8,9 +8,6 @@
 import org.infinispan.commons.marshall.AbstractExternalizer;
 import org.infinispan.commons.marshall.AdvancedExternalizer;
 import org.infinispan.commons.util.Util;
-import org.infinispan.marshall.core.ExternalizerTable;
```
Comment (slaskawi) [inline core/src/test/java/org/infinispan/marshall/core/JBossMarshallerTest.java]: Probably unnecessary change

**Comment by slaskawi** [comment]
May I ask you to squash those commits and update https://github.com/infinispan/infinispan/pull/4388 as well?

**Comment by anistor** [comment]
I've squashed most of them, except the ones that were not related.

File: client/hotrod-client/src/test/java/org/infinispan/client/hotrod/event/EmbeddedCompatContinuousQueryTest.java (hunk 1 above)
Comment (anistor) [inline client/hotrod-client/src/test/java/org/infinispan/client/hotrod/event/EmbeddedCompatContinuousQueryTest.java]: Fixed, using ControlledTimeService.

**Comment by slaskawi** [comment]
Integrating

**Comment by slaskawi** [comment]
Integrated. Thanks @anistor !
